Quick_Deploy/HuggingFaceTransformers/README.md

```diff
@@ -0,0 +1,150 @@
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Deploying a Hugging Face Transformer Models in Triton
+
+The following tutorial demonstrates how to deploy an arbitrary hugging face transformer
+model on the Triton Inference Server using Triton's [Python backend](https://github.com/triton-inference-server/python_backend). For the purposes of this example, three transformer
+models will be deployed:
+- [facebook/opt-125m](https://huggingface.co/facebook/opt-125m) for text generation requests.
+- [gpt2](https://huggingface.co/gpt2) for text generation requests.
+- [distilbert-base-uncased-finetuned-sst-2-english](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) for text classification requests.
+
+These models were selected because of their small size and consistent response quality, 
+enabling the tutorial to run on most devices with sensible results. However, this tutorial
+is also generalizable for much larger models provided sufficient infrastructure. 
+
+*NOTE*: The tutorial is intended to be a reference example only. It is a work in progress.
+
+## Step 1: Build a Model Repository
+
+The first step is to create a model repository that the Triton Inference Server will use
+for inference processing. We can create a model repository from the provided base files by 
+executing the following python script:
+
+```
+python3 create_repository.py
+```
+
+Without specifying any arguments, this script will create a model repository containing
+the details necessary for Triton to load and serve each of the aforementioned models.
+For the purposes of this tutorial, we provide Triton with simple ```config.pbtxt``` files
+for each model that describe:
+- The the backend to use.
+- Model input and output details.
+- Custom parameters to use for execution.
+
+## Step 2: Build a Triton Container Image
```

Comment:
I'm confused about the best practice in shipping a custom Python execution environment, spawning a new image, or using [conda-pack](https://github.com/triton-inference-server/python_backend#creating-custom-execution-environments).

What do you prefer in production-level deployment?

Reply:
@Tabrizian  - thoughts on this?

Reply:
That depends on your goals. If you want to run models with different Python environments on one server, you would [use custom execution environments](https://github.com/triton-inference-server/python_backend#creating-custom-execution-environments). It's more memory-intensive (especially as you increase the number of models), so if you only need one Python version and environment, it would be easier to use an image with all the components needed.

Also, conda-pack can reduce the size of the environment itself (versus using an uncompressed environment), but it then needs to be unpacked at load time, which increases the model load time considerably.

Reply:
Adding to what David mentioned, if you don't want to create a new container that has all the dependencies or you want to isolate the dependencies used by each model custom execution environments are the recommended path.